Initial add of image classification (#24898)
dotnet/docs: docs/machine-learning/resources/tasks.md

You can train a forecasting model with the following algorithm:

<xref:Microsoft.ML.TimeSeriesCatalog.ForecastBySsa%2A>

## Image Classification

> A [supervised machine learning](glossary.md#supervised-machine-learning) task that is used to predict the class (category) of an image. The input is a set of labeled examples. Each label normally starts as text. It is then run through the TermTransform, which converts it to the Key (numeric) type. The output of the image classification algorithm is a classifier, which you can use to predict the class of new images. Examples of image classification scenarios include:

Reviewer: Maybe make a note that Image Classification is an multiclass classification algorithm

* Determining the breed of a dog as a "Siberian Husky", "Golden Retriever", "Poodle", etc.
* Determining if a manufacturing product is defective or not.
* Determining what types of flowers as "Rose", "Sunflower", etc.

### Image classification trainers

You can train an image classification model using the following training algorithms:

* <xref:Microsoft.ML.Vision.ImageClassificationTrainer>

> 

Reviewer: Can we add inputs and outputs

https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.ml.vision.imageclassificationtrainer?branch=main&view=ml-dotnet#input-and-output-columns

## Object Detection

> A [supervised machine learning](glossary.md#supervised-machine-learning) task that is used to predict the class (category) of an image but also gives a bounding box to where that category is within the image. Examples of object detection include:

Reviewer: Consider clarifying that as opposed to image classification, you're classifying multiple objects in an image instead of just one. https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/object-detection-onnx#what-is-object-detection

* Detecting cars, signs, or people on images of a road.
* Detecting defects on images of products.
* Detecting areas of concern on X-Ray images.

> Object detection is currently only available on Azure Machine Learning through [Model Builder](../automate-training-with-model-builder.md).

Reviewer: Consider:

```suggestion
Object detection model training is currently only available in [Model Builder](../automate-training-with-model-builder.md) using Azure Machine Learning.
```
